tests/nougat/nut/git/status_spec.lua

```diff
@@ -60,6 +60,46 @@ describe("nut.git.status", function()
     t.eq(bar:generate(ctx), " ~2 -3 ")
   end)
 
+  it("works for vim-gitgutter", function()
+    package.loaded["gitsigns"] = {}
+
+    bar:add_item(nut.create({
+      prefix = " ",
+      content = {
+        nut.count("added", { prefix = "+", suffix = " " }),
+        nut.count("changed", { prefix = "~", suffix = " " }),
+        nut.count("removed", { prefix = "-", suffix = " " }),
+      },
+    }))
+
+    package.loaded["gitsigns"] = nil
+
+    -- mock data
+    vim.fn["exists"] = function(name)
+      return name == "*GitGutterGetHunkSummary" and 1 or 0
+    end
+    vim.fn["GitGutterGetHunkSummary"] = function()
+      return { 1 }
+    end
+    vim.g["gitgutter_hook_context"] = { bufnr = 0 }
+
+    vim.api.nvim_exec_autocmds("User", { pattern = "GitGutter" })
+    vim.wait(0)
+
+    t.eq(bar:generate(ctx), " +1 ")
```

Comment:
the assertion fails, the result of `generate` gives me `''` empty string.

I think I failed to mock the `vim.fn.GitGutterGetHunkSummary` function returned value.

Reply:
<img width="1346" alt="image" src="https://github.com/MunifTanjim/nougat.nvim/assets/6496887/c179590d-9cc8-4858-b914-6f1b6a23e583">

Reply:
I'll look into it.